app/src/androidTest/java/com/github/sdp_begreen/begreen/fragments/UserPhotoFragmentTest.kt

```diff
@@ -0,0 +1,163 @@
+import com.github.sdp_begreen.begreen.R
+import com.github.sdp_begreen.begreen.models.User
+import com.github.sdp_begreen.begreen.activities.MainActivity
+import com.github.sdp_begreen.begreen.models.ParcelableDate
+import com.github.sdp_begreen.begreen.models.PhotoMetadata
+import junit.framework.TestCase.*
+import org.hamcrest.CoreMatchers
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+import java.util.*
+import kotlin.collections.ArrayList
+
+@RunWith(AndroidJUnit4::class)
+@LargeTest
+class UserPhotoFragmentTest {
+    @get:Rule
+    val activityRule = ActivityScenarioRule(MainActivity::class.java)
+
+    private lateinit var fragment: UserPhotoFragment
+    private lateinit var photoList: List<PhotoMetadata>
+
+    private val ARG_COLUMN_COUNT = "column-count"
+    private val ARG_PHOTO_LIST = "photo-list"
+    private val ARG_IS_FEED = "is-feed"
+
+    @Before
+    fun setup() {
+        // Initialize test data
+        val photos = listOf(
+            PhotoMetadata("1","Look at me cleaning!", ParcelableDate(Date()),User("0",100, "SuperUser69"), "Organique","Wowa je suis incroyable en train de ramasser cette couche usagée pour faire un selfie avec!"), PhotoMetadata("1","Look at me cleaning!", ParcelableDate(
+                Date()
+            ),User("0",100, "SuperUser69"), "Organique","Wowa je suis incroyable en train de ramasser cette couche usagée pour faire un selfie avec!"),
+            PhotoMetadata("1","Look at me cleaning!", ParcelableDate(Date()),User("0",100, "SuperUser69"), "Organique","Wowa je suis incroyable en train de ramasser cette couche usagée pour faire un selfie avec!"),
+            PhotoMetadata("1","Look at me cleaning!", ParcelableDate(Date()),User("0",100, "SuperUser69"), "Organique","Wowa je suis incroyable en train de ramasser cette couche usagée pour faire un selfie avec!"),
+            PhotoMetadata("1","Look at me cleaning!", ParcelableDate(Date()),User("0",100, "SuperUser69"), "Organique","Wowa je suis incroyable en train de ramasser cette couche usagée pour faire un selfie avec!")
+        )
+        photoList = photos
+
+        // Create a new instance of the fragment with test arguments.
+        fragment = UserPhotoFragment.newInstance(2, photos, true)
+    }
+
+    @Test
+    fun onCreateViewFeedWithValidViewReturnsView() {
+        val args = Bundle().apply {
+            putInt(ARG_COLUMN_COUNT, 1)
+            putParcelableArrayList(ARG_PHOTO_LIST, photoList.toCollection(ArrayList()))
+            putBoolean(ARG_IS_FEED, true)
+        }
+
+        // Launch fragment with arguments
+        val scenario = FragmentScenario.launchInContainer(UserPhotoFragment::class.java, args)
+        // Wait for the fragment to be created
+        scenario.onFragment { fragment ->
+            val recyclerView = fragment.view
+            assertNotNull(recyclerView)
+            assertTrue(recyclerView is RecyclerView)
```

Comment:
I recommend you to use hamcrest test to keep coherence with other tests of the project. You can change this line with the following :

`assertThat(recyclerView, instanceOf(RecyclerView::class.java))`